### DISCUSSION TITLE
Resizing Pipeline for GPT-4 Vision

### DISCUSSION THREAD

**Discussion by blue-j** [ImageMagick/ImageMagick#7276] [Help] [answered]
### ImageMagick version

7.1.1-29 Q16-HDRI aarch64

### Operating system, version and so on

macOS 13.6.6

### Description

I'm trying to pre-process images for use with GPT-4 Vision.  The [documentation](https://platform.openai.com/docs/guides/vision) states:

> Images are first scaled to fit within a 2048 x 2048 square, maintaining their aspect ratio. Then, they are scaled such that the shortest side of the image is 768px long.
> 
> Here are some examples demonstrating the above.
> 
> A 1024 x 1024 square image
> 1024 is less than 2048, so there is no initial resize.
> The shortest side is 1024, so we scale the image down to 768 x 768.
> 
> A 2048 x 4096 image
> We scale down the image to 1024 x 2048 to fit within the 2048 square.
> The shortest side is 1024, so we further scale down to 768 x 1536.

I can't seem to crack this.  Presumably we pipe one resize into another?  Any pointers would be gratefully accepted.  : )

- J

**Comment by fmw42** [comment]
The following should do what you want.

`magick image.suffix -resize "2048x2048>" miff:- | magick - -resize "768x768^" result.suffix`

Quotes (or \ escapes) seem to be needed when I paste into my Terminal. But in a script, they may not be needed.

See geometry for > and ^ at https://imagemagick.org/script/command-line-processing.php#geometry

**Comment by snibgo** [comment, marked as answer]
> Presumably we pipe one resize into another?

There is no need for piping. That just adds unneeded complications. Windows syntax:
```
magick in.png -resize "2048x2048^>" -resize "768x768^^" out.png
```
Bash syntax:
```
magick in.png -resize "2048x2048>" -resize "768x768^" out.png
```

**Reply by fmw42** [reply]
OOPS!  Sorry, I forgot the pipe was not needed -- just following what he said.  Dumb!.  Thanks.

**Comment by blue-j** [comment]
thank you so much!!  you're the best.  : )
